core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -103,34 +122,45 @@ public void close() throws IOException {
      */
     public void abort() throws IOException {
         close();
-        tmpFile.delete();
+        Files.deleteIfExists(tmpFile);
     }
 
     public void commit() throws IOException {
         close();
-        if (destFile.exists()) {
-            try {
-                Util.deleteFile(destFile);
-            } catch (IOException x) {
-                tmpFile.delete();
-                throw x;
-            }
+        try {
+            // Try to make an atomic move.
+            Files.move(tmpFile, destFile, StandardCopyOption.ATOMIC_MOVE);
+        } catch (AtomicMoveNotSupportedException e) {
```

Comment:
Looks like this was changed and introduced a :bug:

Reply:
@jtnord I guess you're saying we need to keep the final fallback 

```java
} catch (IOException x) {
       Files.delete(tmpFile);
       throw x;
}
```

right?

Reply:
no it should catch IOEXception and then try to do the move non-atomically.

then if that fails it should throw the resulting exception (with the previous supressed)